src/serverless-stage-destroyer.ts

```diff
@@ -0,0 +1,315 @@
+      return;
+    }
+
+    // Check if any of the stacks have termination protection enabled.
+    await this.checkForTerminationProtection(region, stacksToDestroy);
+
+    /*
+     * Third, show the stacks identified for destroy, and ask the user to confirm before proceeding.
+     *   This underlying function is bypassed when CI=true
+     */
+    if (verify) {
+      this.confirmDestroyCommand(
+        stage,
+        stacksToDestroy.map((a) => `${a.StackName}`)
+      );
+    }
+
+    // Fourth, destroy each stack.
+    for (let i of stacksToDestroy || []) {
+      await this.destroyStack(region, `${i.StackName}`);
+    }
+
+    // Fifth, wait for stacks to be deleted.
+    if (wait) {
+      let waiters = [];
+      for (let i of stacksToDestroy || []) {
+        waiters.push(this.ensureStackIsDeleted(region, `${i.StackName}`));
+      }
+      await Promise.all(waiters);
+    }
+  }
+
+  private async checkForTerminationProtection(region: string, stacks: any[]) {
+    const client = new CloudFormationClient({ region: region });
+    for (const stack of stacks) {
+      let describeStacksCommandResponse = (
+        await client.send(
+          new DescribeStacksCommand({
+            StackName: stack.StackName,
+          })
+        )
+      ).Stacks;
+      let stackDetails = describeStacksCommandResponse
+        ? describeStacksCommandResponse[0]
+        : { EnableTerminationProtection: undefined };
+      if (stackDetails.EnableTerminationProtection) {
+        throw "ERROR:  At least one stack was found to have termination protection enabled.  Refusing to continue.";
+      }
+    }
+  }
+
+  private async ensureStackIsDeleted(region: string, stack: string) {
+    console.log(`Waiting for stack ${stack} to be deleted...`);
+    const client = new CloudFormationClient({ region: region });
+    await waitUntilStackDeleteComplete(
+      {
+        client: client,
+        maxWaitTime: 7200,
+      },
+      {
+        StackName: stack,
+      }
+    );
+  }
+
+  private async getAllStacksForRegion(region: string) {
+    const client = new CloudFormationClient({ region: region });
+    const stacks = [];
+    for await (const page of paginateDescribeStacks({ client }, {})) {
+      stacks.push(...(page.Stacks || []));
+    }
+    return stacks;
+  }
+
+  private async getAllStacksForStage(
+    region: string,
+    stage: string,
+    addFilters?: Tag[]
+  ) {
+    let stacks = await this.getAllStacksForRegion(region);
+    const matchTags = [
+      {
+        Key: "STAGE",
+        Value: stage,
+      },
+    ];
+    matchTags.push(...(addFilters || []));
+    for (let matchTag of matchTags) {
+      stacks = stacks.filter((i) =>
+        i.Tags?.find((j) => j.Key == matchTag.Key && j.Value == matchTag.Value)
+      );
+    }
+    return stacks;
+  }
+
+  private checkForProtectedStage(stage: string) {
+    // Another safeguard against destroying protected stages
+    console.log(`Not protecting prod for stage name: ${stage}`);
```

Comment:
This is the only reason we are bringing in this library's file. We have to override the check otherwise we can't do the conversion in prod.